### PR TITLE
Initial map items/editors work, add map characters tab to script editor

### PR DESCRIPTION
**Pull request by jonko0493**
image
image

Adds map previews with pathing map overlay to the map editor as well as adding a "Map Characters" tab to the script editor which allows for editing the chibis, facing directions, and associated script blocks for the chibis on the map, as well as allowing chibis to be removed. You can also add new chibis and drag and drop them around the map.

### CODE REVIEW

File: src/SerialLoops/Editors/ScriptEditor.cs [hunk 1]
```diff
@@ -186,7 +198,231 @@ private StackLayout GetStartingChibisAddButton(TabPage parent)
...
+                    chibiData.SetObject(chibi, nameof(ChibiItem));
+                    chibiLayout.DoDragDrop(chibiData, DragEffects.Move, chibiIcon, new(8, 16));
+                };
+                chibiLayout.MouseDown += (o, args) =>
+                {
+                    mapDetailsLayout.Items.Clear();
+                    DropDown chibisSelectorDropDown = new();
+                    chibisSelectorDropDown.Items.AddRange(_project.Items.Where(i => i.Type == ItemDescription.ItemType.Chibi).Select(c => new ListItem { Key = c.Name, Text = c.Name }));
+                    chibisSelectorDropDown.SelectedKey = chibi.Name;
+                    DropDown facingDirectionDropDown = new()
```
Comment (WiIIiam278) [inline src/SerialLoops/Editors/ScriptEditor.cs:341]: Maybe for this a 2x2 button array? Can add some nice icons for it, or just use arrows for now, and may be more intuitive. Will also want to reuse in the chibi editor itself

**Review by WiIIiam278 — APPROVED** [review]
extremely cool

File: src/SerialLoops/Editors/ScriptEditor.cs [hunk 2]
```diff
@@ -186,7 +198,231 @@ private StackLayout GetStartingChibisAddButton(TabPage parent)
...
+            {
+                for (int y = 0; y < maps[0].Map.PathingMap[x].Length; y++)
+                {
+                    grid.Add(new((gridZero.X - x * 16 + y * 16) / 2, (gridZero.Y + x * 8 + y * 8) / 2), new Point(x, y));
+                }
+            }
+
+            mapLayout.DragEnter += (obj, args) =>
+            {
+                Console.WriteLine("Hello");
```
[the remark this answers is not in the document]
Comment (jonko0493) [reply src/SerialLoops/Editors/ScriptEditor.cs:268]: this helped me get it working!

File: src/SerialLoops/Editors/ScriptEditor.cs (hunk 1 above)
Comment (jonko0493) [reply src/SerialLoops/Editors/ScriptEditor.cs:341]: I will do this.

**Comment by jonko0493** [comment]
Converting this to a draft for now bc I'm going to try to get drag and drop fully working, but that will take more time than I have rn lol.

File: src/SerialLoops/Editors/ScriptEditor.cs (hunk 1 above)
Comment (jonko0493) [reply src/SerialLoops/Editors/ScriptEditor.cs]: Okay this is a future PR thing bc I couldn't really get it looking nice in this one.

File: src/SerialLoops/Editors/ScriptEditor.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (jonko0493) [reply src/SerialLoops/Editors/ScriptEditor.cs]: Got it working thanks to this. 😎

**Pull request merged**